js/web/lib/wasm/jsep/webgpu/ops/range.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {DataType} from '../../../wasm-common';
+import {TensorView} from '../../tensor';
+import {ComputeContext, GpuDataType, ProgramInfo, ProgramMetadata} from '../types';
+
+import {getWgslValueType, outputVariable, ShaderHelper} from './common';
+
+const validateInputs = (inputs: readonly TensorView[]): void => {
+  if (!inputs || inputs.length !== 3) {
+    throw new Error('Range requires 3 input.');
+  }
+
+  if (inputs[0].dims.length !== 0 || inputs[1].dims.length !== 0 || inputs[2].dims.length !== 0) {
+    throw new Error('Range requires scalar input.');
+  }
+
+  if (inputs[0].dataType !== inputs[1].dataType || inputs[1].dataType !== inputs[2].dataType) {
+    throw new Error('Range requires all inputs have the same data type.');
+  }
+};
+
+const validateInputsContent = (start: number, limit: number, delta: number): void => {
+  const sameStartLimit = start === limit;
+  const increasingRangeNegativeStep = start < limit && delta < 0;
+  const decreasingRangePositiveStep = start > limit && delta > 0;
+
+  if (sameStartLimit || increasingRangeNegativeStep || decreasingRangePositiveStep) {
+    throw new Error('Range these inputs are invalid.');
+  }
+};
+
+const createRangeProgramInfo =
+    (metadata: ProgramMetadata, start: number, limit: number, delta: number, dataType: DataType): ProgramInfo => {
+      const numElements = Math.abs(Math.ceil((limit - start) / delta));
+      const outputShape: number[] = [numElements];
+      const outputSize = numElements;
+
+      const output = outputVariable('output', dataType, outputShape);
+      const mappedType = getWgslValueType(dataType, 1);
+      const wgslType = typeof mappedType === 'string' ? mappedType : mappedType[1];
```

Comment:
I think you still can use `const wgslType = inputVariable('input', context.inputs[0].dataType, context.inputs[0].dims).type.value;` to get the type.

Reply:
Here I only want to get the wgsl type, if I use `inputVariable` here, which will create a `IndicesHelper` (a big object), and run huge unnecessary code here https://github.com/microsoft/onnxruntime/blob/d5c565156db023af400ff4e18780bdb2862eb343/js/web/lib/wasm/jsep/webgpu/ops/common.ts#L258. I think it is not reasonable, do you think so?

Reply:
in latest main branch, common.ts exposes `tensorTypeToWsglStorageType()` function for the usage of this. Please resolve conflicts and use this function.

Reply:
Also, in general using IndicesHelper is recommended. If we already using IndicesHelper for input/output, we can use it to access the storage type.

in this case, you already defined `output`, so use `output.type.storage` may be the good option here.